### ISSUE TITLE
[🐛 Bug]: Selenium Manager Should honor full browser version 

### ISSUE THREAD

**Issue by titusfortner** [SeleniumHQ/selenium#13419]
### What happened?

If a user specifies a specific patch version of the browser and it is available, Selenium should download it.

### How can we reproduce the issue?

```shell
"selenium-manager", "--browser", "chrome", "--browser-version", "120.0.6099.71", "--language-binding", "ruby", "--output", "mixed", "--debug"
```

### Relevant log output

```shell
DEBUG	chromedriver not found in PATH
DEBUG	chrome detected at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
DEBUG	Running command: /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --version
DEBUG	Output: "Google Chrome 120.0.6099.199 "
DEBUG	Detected browser: chrome 120.0.6099.199
DEBUG	Required driver: chromedriver 120.0.6099.109
DEBUG	Discovering versions from https://googlechromelabs.github.io/chrome-for-testing/known-good-versions-with-downloads.json
DEBUG	Downloading chromedriver 120.0.6099.109 from https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/120.0.6099.109/mac-x64/chromedriver-mac-x64.zip
INFO	Driver path: /Users/titusfortner/.cache/selenium/chromedriver/mac-x64/120.0.6099.109/chromedriver
INFO	Browser path: /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
```

### Operating System

Mac

### Selenium version

sm_mixed_output branch

### What are the browser(s) and version(s) where you see this issue?

n/a

### What are the browser driver(s) and version(s) where you see this issue?

n/a

### Are you using Selenium Grid?

n/a

**Comment by diemol** [comment]
The decision to implement this was made during the Selenium Dev Summit in May 2024.

**Comment by shbenzer** [comment]
Could someone advise me on this issue? I'm trying to learn exactly how selenium manager downloads the required binaries, but I cannot figure out where selenium manager is utilizing the get_chrome_milestone() method to pull the available versions

**Comment by bonigarcia** [comment]
I created a PR for this: #14619.

@shbenzer For implementing it, the general logic for browser discovery needed to be changed. Also, the specific logic for filtering the the releases from the CfT endpoints was changed.

**Comment by shbenzer** [comment]
Awesome!

**Comment by bonigarcia** [comment]
Closed via #14619.
